### PR TITLE
Updates to docs crawler

### PR DESCRIPTION
**Pull request by ofermend**
- Changed docusaurus crawler to use playwright and crawling of docs instead of using the repo to avoid edge cases
- Improved index_url to support additional file types

### CODE REVIEW

File: config/langchain-docusaurus-search.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+vectara:
```
Comment (ofermend) [inline config/langchain-docusaurus-search.yaml:1]: Added configuration for crawling all of langchain docs

File: config/vectara-docusaurus-search.yaml [hunk 2]
```diff
@@ -7,5 +7,5 @@ crawling:
   crawler_type: docusaurus
 
 docusaurus_crawler:
-  docs_repo: "https://github.com/vectara/vectara-docs"
   docs_homepage: "https://docs.vectara.com/docs"
+  url_regex: [".*vectara.com/docs.*"]
```
Comment (ofermend) [inline config/vectara-docusaurus-search.yaml:11]: Parameters for docs crawl now support regex patterns

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-requests==2.28.2
+requests==2.31.0
```
Comment (ofermend) [inline requirements.txt:1]: Upgrade to fix security vulnerability

**Comment by justinhayes** [comment]
This all looks good to me. Although I don't have any experience with playwright, so for that part of it my review isn't worth all that much.

File: core/indexer.py [hunk 4]
```diff
@@ -176,7 +217,9 @@ def index_url(self, url: str, metadata: dict) -> bool:
             soup = BeautifulSoup(content, 'html.parser')
             title = soup.title.string if soup.title else 'No Title'
             elements = partition_html(text=content)
-            elements = [str(e) for e in elements if type(e)==us.documents.elements.NarrativeText]
+
+        # only capture the text elements (as str) and ignore others
```
Comment (eskibars) [inline core/indexer.py:221]: Can you provide a bit more detail on why this needs to be done?

**Review by eskibars — APPROVED** [review]
LGTM.  Thanks!

File: core/indexer.py (hunk 4 above)
Comment (ofermend) [reply core/indexer.py:221]: The idea is to only take narrative text elements and the str(e) converts them to an actual string.
See https://unstructured-io.github.io/unstructured/getting_started.html#document-elements for other element types. We can refine this later, but wanted to keep it simple to start.

**Pull request merged**